### PR TITLE
[LMS 636] - [API/ Integration] Settings tab in learning path detail page

### PR DESCRIPTION
**Pull request by elyric-sun-asterisk-dev**
## Issue Link
https://framgiaph.backlog.com/view/LMS-636

## Defintion of Done

- [x] Edit name
- [x] Edit description
- [x] Edit or delete courses from a learning path
- [x] Delete learning path

## Steps to reproduce
`docker exec -it lms-django bash`
`pipenv shell`
`pipenv run migrate`

## Affected Components / Functionalities / Page
api/app_sph_lms/admin.py
apilapp_sph_lms/api/serializer/learning_path serializer.py
api/app_sph_lms/api/urls.py
api/app_sph_lms/api/view/learning_path_view.py
api/app_sph_lms/migrations/0001_initial.py
api/app_sph_lms/migrations/0002_remove course status course_category_and more.py
api/app_sph_lms/migrations/0003_alter_lesson_link.py
api/app_sph_lms/migrations/0004 alter_lesson_order.py
api/app_sph_lms/migrations/0005 alter_lesson_order.py
api/app_sph_lms/migrations/0006_learningpath_rename_img_path_course_image_and_more.py
api/app_sph_lms/migrations/0007 alter_learningpath_author.py
api/app_sph_lms/models.py
client/features/learning-path/learningPathSlice.ts
client/services/learningPathAPl.ts
client/src/pages/trainer/learning-paths/[id]/index.tsx
client/src/pages/trainer/learning-paths/index.tsx
client/src/sections/learning-paths/ContentSection/index.tsx
client/src/sections/learning-paths/settingsSection/EditSettingsButton.tsx
client/src/sections/learning-paths/settingsSectionTlndex.tsx
client/src/shared/components/Card/LearningPathCardflndex.tsx
client/src/shared/utils/interface.ts

## Screenshots
image

### CODE REVIEW

**Review by FranzDelosSantos — COMMENTED** [review]
I also encountered these errors, there is nothing wrong in the master branch so I assume this is from you latest changes.

Error in creating learning path
image

Archive learning path
image

**Review by FranzDelosSantos — COMMENTED** [review]
There are also some npm run build errors, pls fix it 
image

File: api/app_sph_lms/api/serializer/learning_path_serializer.py [hunk 1]
```diff
@@ -1,202 +1,111 @@
...
         queryset=Category.objects.all(),
         many=True,
+        required=False
     )
     courses = serializers.ListField(
-      child=LearningPathCourseSerializer(),
-      write_only=True
+        child=LearningPathCourseSerializer(), write_only=True
     )
-    course_count = serializers.SerializerMethodField()
```
[the remark this answers is not in the document]
Comment (elyric-sun-asterisk-dev) [reply api/app_sph_lms/api/serializer/learning_path_serializer.py:31]: I didn't change anything related to that. I only modified these two files only when fixing some errors on my end yesterday 🤔🙇 
image

File: api/app_sph_lms/api/serializer/learning_path_serializer.py [hunk 2]
```diff
@@ -1,202 +1,111 @@
...
 from app_sph_lms.models import (Category, Course, LearningPath,
-                                LearningPathCourse, User)
+                                LearningPathCourse)
 from rest_framework import serializers
-from rest_framework.pagination import PageNumberPagination
-
-
-class CustomPagination(PageNumberPagination):
-    page_size = 10
-    page_size_query_param = 'page_size'
```
[the remark this answers is not in the document]
Comment (elyric-sun-asterisk-dev) [reply api/app_sph_lms/api/serializer/learning_path_serializer.py:13]: I didn't remove one of that but we can put it back 🙇

File: api/app_sph_lms/api/serializer/learning_path_serializer.py [hunk 3]
```diff
@@ -1,202 +1,111 @@
...
-    learners = serializers.SerializerMethodField()
+class ExtendedCourse(CourseSerializer):
+    order = serializers.SerializerMethodField()
 
-    class Meta:
-        model = LearningPath
-        fields = ['learners']
-
-    def get_learners(self, obj):
-        is_enrolled = self.context[
```
[the remark this answers is not in the document]
Comment (elyric-sun-asterisk-dev) [reply api/app_sph_lms/api/serializer/learning_path_serializer.py:125]: I'm not sure to that for now, maybe we can have a huddle for that issue 🙇

**Review by FranzDelosSantos — APPROVED** [review]
## Approved
Iteration: 2
Comment : 7
Testing Time : 47min

**Review by Melanthonyando — APPROVED** [review]
### Approved

Proof :
[Untitled_ Jun 30 2023 3_51 PM.webm](https://github.com/framgia/sph-lms/assets/110366151/54192bd9-799b-4c0f-8e31-0791a9ffd5ab)

Iteration : 0
Comment: 0
Testing Time: 45 mins.

**Pull request merged**